codalab/lib/download_manager.py

```diff
@@ -104,7 +104,8 @@ def _get_target_info_within_bundle(self, target, depth):
                 )
             )
         elif bundle_state != State.RUNNING:
-            bundle_path = self._bundle_store.get_bundle_location(target.bundle_uuid)
+            bundle = self._bundle_model.get_bundle(target.bundle_uuid)
```

Comment:
I think it's okay. We already make another DB call with `get_bundle_state` earlier in this function, so we're only linearly increasing the runtime.

I'd rather prioritize keeping the changes simple and readable for now. In the future, though, we probably could optimize all calls to get_bundle_location for DB call speed, though. (using a db query similar to the one written in `BundleModel.get_bundle_names`)?

Reply:
`get_bundle` I think could be much heavier because it gets all the metadata too.  Can you profile to see what is the impact on performance?  Some people use CodaLab in a for loop, where the difference matters.  Either way, please make a comment that only the location is needed.

Reply:
I think I'd rather just go ahead and optimize it rather than profile it 😉

Reply:
It also appears we're calling `get_bundle` multiple times in a loop (for things such as resolving parent dependencies), so we should change this.